site/docs/features/gc-internals.md

```diff
@@ -0,0 +1,182 @@
+# Nessie GC Internals
```

Comment:
Do we really want this on the project site? Would a source repo `.md` file be find for "internal" stuff?

Reply:
I'm open for both variants.

Reply:
It seems to me that keeping internal docs in src tree is preferable, but I do not insist.